modules/exporter-common/src/main/scala/io/janstenpickle/trace4cats/export/ExportRetryConfig.scala

```diff
@@ -1,9 +1,17 @@
 package io.janstenpickle.trace4cats.`export`
 
+import cats.kernel.Eq
+
 import scala.concurrent.duration._
 
 case class ExportRetryConfig(
   delay: FiniteDuration = 500.millis,
   nextDelay: FiniteDuration => FiniteDuration = _ + 100.millis,
   maxAttempts: Int = 100
 )
+
+object ExportRetryConfig {
+  implicit val eq: Eq[ExportRetryConfig] = Eq.by { config =>
+    (config.delay, config.nextDelay(config.delay), config.maxAttempts)
```

Comment:
Hm... This is riskier, I think. This instance may lead to false `eqv`. And the new instance won't be created when needed, e.g. when only the function has changed but its `next` coincides with the `next` of the previous function.

Reply:
How can we compare 2 functions? Maybe by probing more than 2 points? But how many?

Reply:
What are the downsides of swapping resources when the config is not changed?
What are the downsides of NOT swapping when needed?

Reply:
Let's make ExportRetryConfig a sealed trait of 3 cases:
- Constant
- Exponential
- Fibonacci